src/io/flutter/view/TogglePlatformAction.java

```diff
@@ -36,34 +78,58 @@ public void update(@NotNull AnActionEvent e) {
     app.hasServiceExtension(extensionDescription.getExtension(), (enabled) -> {
       e.getPresentation().setEnabled(app.isSessionActive() && enabled);
     });
+
+    e.getPresentation().putClientProperty(SELECTED_PROPERTY, selected);
+
+    if (currentValueSubscription == null) {
+      currentValueSubscription =
+        app.getVMServiceManager().getServiceExtensionState(extensionDescription.getExtension()).listen((state) -> {
+          this.setSelected(e, state);
+        }, true);
+    }
+  }
+
+  @Override
+  public void dispose() {
+    if (currentValueSubscription != null) {
+      currentValueSubscription.dispose();
+      currentValueSubscription = null;
+    }
   }
 
   @Override
   public void perform(AnActionEvent event) {
     if (app.isSessionActive()) {
-      app.togglePlatform().thenAccept(isAndroid -> {
-        if (isAndroid == null) {
-          return;
-        }
-
-        app.togglePlatform(!isAndroid).thenAccept(isNowAndroid -> {
-          if (app.getConsole() != null && isNowAndroid != null) {
-            isCurrentlyAndroid = isNowAndroid;
-
-            app.getConsole().print(
-              FlutterBundle.message("flutter.view.togglePlatform.output",
-                                    isNowAndroid ? "Android" : "iOS"),
-              ConsoleViewContentType.SYSTEM_OUTPUT);
-
-            app.getVMServiceManager().setServiceExtensionState(
-              extensionDescription.getExtension(),
-              true,
-              isNowAndroid
-              ? extensionDescription.getDisabledValue()
-              : extensionDescription.getEnabledValue());
-          }
-        });
-      });
+      app.togglePlatform(platformTarget.name());
+
+      app.getVMServiceManager().setServiceExtensionState(
+        extensionDescription.getExtension(),
+        true,
+        platformTarget.name());
+
+      app.getConsole().print(
```

Comment:
Does this print still make sense now that it is unambiguous what the platform override got set to?